src/app/router/Welcome/Welcome.tsx

```diff
@@ -71,6 +85,25 @@ function App() {
   const location = useLocation();
   const routesElement = useRoutes(routes);
 
+  const [languageChanged, setLanguageChanged] = useState(false);
+  i18n.on("languageChanged", () => {
+    connectorRoutes = getConnectorRoutes();
+    routes = getRoutes(connectorRoutes);
+
+    // Update name to new language only, don't update status
+    const tempSteps: Step[] = [];
+    for (let i = 0; i < routes.length; i++) {
+      tempSteps.push({
+        id: routes[i].name,
+        status: steps[i].status,
```

Comment:
I'm open if anyone has ideas on how to simplify this, but I think either way we'd have to do something like this to update the strings on language change